geostore/check_stac_metadata/utils.py

```diff
@@ -150,7 +156,7 @@ def process_assets(self) -> None:
             assert isinstance(asset_url, str)
 
             self.processing_assets_model(
-                hash_key=self.hash_key,
+                hash_key=get_hash_key(self.dataset_id, self.new_version_id),
```

Comment:
Looks like `get_hash_key` is called for each metadata file and asset. We probably want to call it in the constructor instead.